recipes/libglvnd/meta.yaml

```diff
@@ -0,0 +1,123 @@
+{% set version = "1.7.0" %}
+
+package:
+  name: libglvnd-split
+  version: {{ version }}
+
+source:
+  url: https://gitlab.freedesktop.org/glvnd/libglvnd/-/archive/v{{ version }}/libglvnd-v{{ version }}.tar.bz2
+  sha256: d0e1925a3c9aee0143b8c181ac31d5637c8faa081759c277b8e16c7075612c11
+  patches:
+    - libglvnd_look_for_egl_icd_in_system_location.patch
+
+build:
+  skip: true  # [not linux]
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ stdlib('c') }}
+    - meson
+    - pkg-config
+    - ninja
+  host:
+    - xorg-libx11
+    - xorg-libxext
+    - xorg-xextproto
+    - xorg-glproto
+
+outputs:
+  - name: libglvnd
+    script: install_libglvnd.sh
+    requirements:
+      build:
+        - {{ compiler('c') }}
+        - {{ stdlib('c') }}
+        - meson
+        - pkg-config
+        - ninja
+      host:
+        - xorg-libx11
+        - xorg-libxext
+        - xorg-xextproto
+        - xorg-glproto
```

Comment:
Are these only needed for libGLX ?

Reply:
libGLX, libOpenGL, and libGL.
libEGL if rendering to X11.

Reply:
https://gitlab.freedesktop.org/glvnd/libglvnd/-/blob/master/meson.build?ref_type=heads#L101

Reply:
Because glvnd is merely a broker between OpenGL applications and the GPU driver libraries, each library file is in the range of kilobytes.

Reply:
It looks like only libGLX.so links to the X11 libraries, so I think it would be good to split that out. (conda package named libglx) Is libGLX commonly used in conda-forge packages?

Reply:
And a `libGL` too since `libGL.so` links to `libGLX.so`

Reply:
Then if so, divide every library into its own components like Fedora.

Reply:
https://fedora.pkgs.org/40/fedora-x86_64/libglvnd-devel-1.7.0-4.fc40.x86_64.rpm.html
https://fedora.pkgs.org/40/fedora-x86_64/libglvnd-opengl-1.7.0-4.fc40.x86_64.rpm.html
https://fedora.pkgs.org/40/fedora-x86_64/libglvnd-glx-1.7.0-4.fc40.x86_64.rpm.html
https://fedora.pkgs.org/40/fedora-x86_64/libglvnd-gles-1.7.0-4.fc40.x86_64.rpm.html
https://fedora.pkgs.org/40/fedora-x86_64/libglvnd-egl-1.7.0-4.fc40.x86_64.rpm.html
https://fedora.pkgs.org/40/fedora-x86_64/libglvnd-core-devel-1.7.0-4.fc40.x86_64.rpm.html
https://fedora.pkgs.org/40/fedora-x86_64/libglvnd-1.7.0-4.fc40.x86_64.rpm.html

Reply:
And the hardest part here will be handling the run exports. How?

Reply:
> Is libGLX commonly used in conda-forge packages?

I bet it will have millions of downloads.

Reply:
> It looks like only libGLX.so links to the X11 libraries

https://gitlab.freedesktop.org/glvnd/libglvnd/-/blob/master/src/EGL/libegl.c#L36
https://gitlab.freedesktop.org/glvnd/libglvnd/-/blob/master/meson.build?ref_type=heads#L101

What do you think? I'm asking because I'm not sure.

Reply:
There are dynamic imports.

Reply:
I pushed a WIP commit. What do you think?

Reply:
> I pushed a WIP commit. What do you think?

You need to put in libGL.so as well. They should be bundled as in https://fedora.pkgs.org/40/fedora-x86_64/libglvnd-glx-1.7.0-4.fc40.x86_64.rpm.html.

Reply:
https://packages.debian.org/bookworm/libgl1
https://fedora.pkgs.org/40/fedora-x86_64/libglvnd-glx-1.7.0-4.fc40.x86_64.rpm.html

`libGL.so` depends on `libGLX.so`. There is working around that. And what the OpenGL packages want typically is `libGL.so`, so I fear you are making things a lot complicated for limited earnings.

Reply:
Either way, `libGL.so` is going into `libglx`, together with `libGLX.so`.

Reply:
```
The code in the src/ directory is organized as follows:

* GLX/ contains code for libGLX, the GLX window-system API library.
* EGL/ contains code for libEGL, the EGL window-system API library.
* GLdispatch/ contains code for libGLdispatch, which is responsible for dispatching OpenGL functions to the correct vendor library based on the current EGL or GLX rendering context. This implements the guts of the GL API libraries. Most of the dispatch code is based on Mesa's glapi.
* OpenGL/, GLESv1/, and GLESv2/ contain code to generate libOpenGL.so, libGLESv1_CM.so, and libGLESv2.so, respectively. All three are merely wrapper libraries for libGLdispatch. Ideally, these could be implemented via ELF symbol filtering, but in practice they need to be implemented manually. See the Issues section for details on why this is the case.
* GL/ contains code for libGL. This is a wrapper around libGLdispatch and libGLX.
* util/ contains generic utility code.
```

Reply:
> I pushed a WIP commit. What do you think?

I think that one thing unnecessary is dividing the non-SOVER `libGLdispatch.so` in the devel packages. Won't that complicate the symlinking even more?